### PR TITLE
add ends with option

### PR DESCRIPTION
**Pull request by leandrorodrigueszup**
Signed-off-by: LucasSales <lucas.sales@zup.com.br>

### CODE REVIEW

File: moove/application/src/main/kotlin/io/charlescd/moove/application/circle/request/CreateCircleRequest.kt [hunk 1]
```diff
@@ -83,6 +83,7 @@ data class NodePart(
         LOWER_THAN,
         LESS_THAN_OR_EQUAL,
         STARTS_WITH,
+        ENDS_WITH,
```
Comment (thallesfreitaszup) [inline moove/application/src/main/kotlin/io/charlescd/moove/application/circle/request/CreateCircleRequest.kt:86]: Missing MATCHES expression here

File: ui/src/modules/Circles/Segments/conditional.options.tsx [hunk 2]
```diff
@@ -21,5 +21,6 @@ export default [
   { value: 'LOWER_THAN', label: 'Lower than' },
   { value: 'LESS_THAN_OR_EQUAL', label: 'Lower or equal to' },
   { value: 'STARTS_WITH', label: 'Starts with' },
-  { value: 'NOT_EQUAL', label: 'Not equal' }
+  { value: 'NOT_EQUAL', label: 'Not equal' },
```
Comment (thallesfreitaszup) [inline ui/src/modules/Circles/Segments/conditional.options.tsx:24]: Missing MATCHES expression here

**Review by thallesfreitaszup — CHANGES_REQUESTED** [review]

File: moove/application/src/main/kotlin/io/charlescd/moove/application/circle/request/CreateCircleRequest.kt (hunk 1 above)
Comment (leandrorodrigueszup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/circle/request/CreateCircleRequest.kt:86]: Done

File: ui/src/modules/Circles/Segments/conditional.options.tsx (hunk 2 above)
Comment (leandrorodrigueszup) [reply ui/src/modules/Circles/Segments/conditional.options.tsx:24]: Done

**Review by thallesfreitaszup — APPROVED** [review]

File: circle-matcher/src/main/java/io/charlescd/circlematcher/infrastructure/Constants.java [hunk 3]
```diff
@@ -47,7 +47,7 @@ private Constants() {
 
     public static final String EQUALS_EXPRESSION = "%s.indexOf(%s) != -1";
 
-    public static final String NOT_EQUALS_EXPRESSION = "%s.indexOf(%s) >= 0";
+    public static final String NOT_EQUALS_EXPRESSION = "%s.indexOf(%s) == -1";
```
Comment (lucasbfernandes) [inline circle-matcher/src/main/java/io/charlescd/circlematcher/infrastructure/Constants.java:50]: Why not compare strings using the JS inequality operator?

%s !== %s

**Review by lucasbfernandes — CHANGES_REQUESTED** [review]

**Review by lucasbfernandes — APPROVED** [review]

**Comment by eliasalveszup** [comment]
Canceled due to another PR #649

**Pull request closed without merging**